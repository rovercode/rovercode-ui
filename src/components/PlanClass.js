import React from 'react';
import {
  Grid,
  InputAdornment,
  Link,
  TextField,
  Typography,
} from '@material-ui/core';
import { CheckCircle, School } from '@material-ui/icons';
import { withStyles } from '@material-ui/core/styles';
import { hot } from 'react-hot-loader';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';
import moment from 'moment';

const CodeInput = withStyles((theme) => ({
  root: {
    paddingRight: theme.spacing(1),
    paddingLeft: theme.spacing(1),
  },
}))(TextField);

const LengthDisplay = withStyles((theme) => ({
  root: {
    color: theme.palette.text.secondary,
  },
}))(Typography);

const SuccessIcon = withStyles((theme) => ({
  root: {
    color: theme.palette.success.main,
  },
}))(CheckCircle);

const PlanClass = ({
  active,
  error,
  expires,
  maxLength,
  refreshSession,
  upgradeSubscription,
  userId,
}) => {
  const [length, setLength] = React.useState(0);

  const handleCodeChange = (event) => {
    setLength(event.target.value.length);
    if (event.target.value.length === maxLength) {
      upgradeSubscription(userId, event.target.value).then(refreshSession);
    }
  };

  const generateAdornment = () => {
    if (active) {
      return (
        <InputAdornment position="start">
          <SuccessIcon />
        </InputAdornment>
      );
    }

    if (length === 0) {
      return (null);
    }

    return (
      <LengthDisplay variant="subtitle2">
        {`${length}/${maxLength}`}
      </LengthDisplay>
    );
  };

  return (
    <Grid container direction="column" justify="center" alignItems="stretch">
      {
        !active ? (
          <>
            <Grid item>
              <Typography variant="subtitle2" align="center">
                <FormattedMessage
                  id="app.plan_class.code1"
                  description="Prompts the user to enter class code"
                  defaultMessage="If you have a class code, enter"
                />
              </Typography>
            </Grid>
            <Grid item>
              <Typography variant="subtitle2" align="center">
                <FormattedMessage
                  id="app.plan_class.code2"
                  description="Prompts the user to enter class code"
                  defaultMessage="it below to join your class plan."
                />
              </Typography>
            </Grid>
          </>
        ) : (
          <Grid item>
            <Typography variant="subtitle2" align="center">
              <FormattedMessage
                id="app.plan_class.expires"
                description="Displays the date when the subscription expires"
                defaultMessage="Your class plan expires:"
              />
              {` ${moment.unix(expires).format('M/D/YYYY')}`}
            </Typography>
          </Grid>
        )
      }
      <Grid item>
        <CodeInput
          variant="outlined"
          size="small"
          placeholder="Class Code"
          error={error !== null}
          helperText={error}
          onChange={handleCodeChange}
          InputProps={{
            startAdornment: (
              <InputAdornment position="start">
                <School />
              </InputAdornment>
            ),
            endAdornment: generateAdornment(),
          }}
          inputProps={{ // eslint-disable-line react/jsx-no-duplicate-props
            maxLength,
            readOnly: active,
          }}
        />
      </Grid>
      {
        !active ? (
          <>
            <Grid item>
              <hr />
            </Grid>
            <Grid item>
              <Typography variant="subtitle2" align="center">
                <FormattedMessage
                  id="app.plan_class.contact1"
                  description="Prompts the user to contact for sign up"
                  defaultMessage="Contact us to setup a class plan"
                />
              </Typography>
            </Grid>
            <Grid item>
              <Typography variant="subtitle2" align="center">
                <FormattedMessage
                  id="app.plan_class.contact2"
                  description="Prompts the user to contact for sign up"
                  defaultMessage="for your student group."
                />
              </Typography>
            </Grid>
            <Grid item container direction="column" alignItems="center">
              <Grid item>
                <Link href="mailto:sales@rovercode.com" color="secondary">
                  sales@rovercode.com
                </Link>
              </Grid>
            </Grid>
          </>
        ) : (null)
      }
    </Grid>
  );
};

PlanClass.defaultProps = {
  active: false,
  error: null,
  expires: 0,
  maxLength: 8,
};

PlanClass.propTypes = {
  active: PropTypes.bool,
  error: PropTypes.string,
  expires: PropTypes.number,
  maxLength: PropTypes.number,
  refreshSession: PropTypes.func.isRequired,
  upgradeSubscription: PropTypes.func.isRequired,
  userId: PropTypes.number.isRequired,
};

export default hot(module)(PlanClass);
