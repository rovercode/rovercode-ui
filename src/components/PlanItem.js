import React from 'react';
import { Box, Typography } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import { hot } from 'react-hot-loader';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';

const Banner = withStyles((theme) => ({
  root: {
    backgroundColor: theme.palette.secondary.main,
    color: theme.palette.getContrastText(theme.palette.secondary.main),
    borderRadius: '4px 4px 0px 0px',
    maxWidth: '90%',
    marginLeft: 'auto',
    marginRight: 'auto',
  },
}))(Typography);

const PlanItem = ({ active, children, title }) => (
  <>
    {
    active === true ? (
      <Banner variant="subtitle2" align="center">
        <FormattedMessage
          id="app.plan_item.your_plan"
          description="Identifier for the subscription plan"
          defaultMessage="Your plan"
        />
      </Banner>
    ) : (null)
  }
    <Box p={2} border={1} borderRadius="borderRadius" boxShadow={2} borderColor="grey.300">
      <Typography variant="h5" align="center">
        {title}
      </Typography>
      {children}
    </Box>
  </>
);

PlanItem.defaultProps = {
  active: false,
};

PlanItem.propTypes = {
  active: PropTypes.bool,
  children: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
};

export default hot(module)(PlanItem);
