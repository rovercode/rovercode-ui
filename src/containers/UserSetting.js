import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { hot } from 'react-hot-loader';
import { withCookies, Cookies } from 'react-cookie';
import { editUserPassword, editUserUsername, refreshSession } from '@/actions/user';
import { checkAuthError, authHeader } from '@/actions/auth';
import { fetchSubscription, upgradeSubscription } from '@/actions/subscription';
import UserSetting from '@/components/UserSetting';

const mapStateToProps = ({ user, subscription }) => ({ user, ...subscription });
const mapDispatchToProps = (dispatch, { cookies }) => ({
  editUserUsername: (username) => dispatch(editUserUsername(username, authHeader(cookies)))
    .catch(checkAuthError(dispatch)),
  editUserPassword: (password) => dispatch(editUserPassword(password, authHeader(cookies)))
    .catch(checkAuthError(dispatch)),
  fetchSubscription: (id) => dispatch(fetchSubscription(id, authHeader(cookies))),
  refreshSession: () => dispatch(refreshSession(cookies)).catch(checkAuthError(dispatch)),
  upgradeSubscription: (id, accessCode) => dispatch(
    upgradeSubscription(id, accessCode, authHeader(cookies)),
  ),
});

const UserSettingContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(UserSetting);

UserSettingContainer.propTypes = {
  cookies: PropTypes.instanceOf(Cookies).isRequired,
};

export default hot(module)(withCookies(UserSettingContainer));
