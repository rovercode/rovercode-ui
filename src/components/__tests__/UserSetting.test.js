import React from 'react';
import { Redirect } from 'react-router-dom';
import { TextField } from '@material-ui/core';
import { Alert } from '@material-ui/lab';
import PlanClass from '../PlanClass';
import PlanItem from '../PlanItem';
import UserSetting from '../UserSetting';

let editUserPassword;
let editUserUsername;
let fetchSubscription;
let refreshSession;
let upgradeSubscription;

describe('The UserSetting component', () => {
  beforeEach(() => {
    editUserPassword = jest.fn().mockResolvedValue();
    editUserUsername = jest.fn().mockResolvedValue();
    fetchSubscription = jest.fn().mockResolvedValue();
    refreshSession = jest.fn().mockResolvedValue();
    upgradeSubscription = jest.fn().mockResolvedValue();
  });

  test('renders on the page with no errors', () => {
    const user = {
      user_id: 1,
      username: 'testuser',
      email: 'test@example.com',
      isSocial: false,
    };
    const wrapper = shallowWithIntl(
      <UserSetting
        user={user}
        editUserPassword={editUserPassword}
        editUserUsername={editUserUsername}
        fetchSubscription={fetchSubscription}
        refreshSession={refreshSession}
        upgradeSubscription={upgradeSubscription}
        isFetching
      />,
    ).dive().dive().dive();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find({ type: 'password' }).length).toBe(2);
  });

  test('hides password edit when social user', () => {
    const user = {
      user_id: 1,
      username: 'testuser',
      email: 'test@example.com',
      isSocial: true,
    };
    const wrapper = shallowWithIntl(
      <UserSetting
        user={user}
        editUserPassword={editUserPassword}
        editUserUsername={editUserUsername}
        fetchSubscription={fetchSubscription}
        refreshSession={refreshSession}
        upgradeSubscription={upgradeSubscription}
        isFetching
      />,
    ).dive().dive().dive();
    expect(wrapper.find({ type: 'password' }).exists()).toBe(false);
  });

  test('saves user username', (done) => {
    const user = {
      user_id: 1,
      username: 'testuser',
      email: 'test@example.com',
      isSocial: false,
    };
    const wrapper = shallowWithIntl(
      <UserSetting
        user={user}
        editUserPassword={editUserPassword}
        editUserUsername={editUserUsername}
        fetchSubscription={fetchSubscription}
        refreshSession={refreshSession}
        upgradeSubscription={upgradeSubscription}
        isFetching
      />,
    ).dive().dive().dive();

    wrapper.find(TextField).first().simulate('change', {
      target: {
        name: 'username',
        value: 'newuser',
      },
    });
    wrapper.update();
    wrapper.instance().saveUserUsername({ preventDefault: jest.fn() }).then(() => {
      expect(wrapper.state('saveSuccess')).toBe(true);
      expect(wrapper.state('usernameError')).toBeNull();
      expect(wrapper.find(Redirect).exists()).toBe(true);
      expect(wrapper.find(Redirect).prop('to')).toBe('/accounts/login');
      expect(editUserPassword).not.toHaveBeenCalled();
      expect(editUserUsername).toHaveBeenCalledWith('newuser');
      done();
    });
  });

  test('saves user password', (done) => {
    const user = {
      user_id: 1,
      username: 'testuser',
      email: 'test@example.com',
      isSocial: false,
    };
    const wrapper = shallowWithIntl(
      <UserSetting
        user={user}
        editUserPassword={editUserPassword}
        editUserUsername={editUserUsername}
        fetchSubscription={fetchSubscription}
        refreshSession={refreshSession}
        upgradeSubscription={upgradeSubscription}
        isFetching
      />,
    ).dive().dive().dive();

    wrapper.find(TextField).at(1).simulate('change', {
      target: {
        name: 'password1',
        value: 'password',
      },
    });
    wrapper.find(TextField).at(2).simulate('change', {
      target: {
        name: 'password2',
        value: 'password',
      },
    });
    wrapper.update();
    wrapper.instance().saveUserPassword({ preventDefault: jest.fn() }).then(() => {
      expect(wrapper.state('saveSuccess')).toBe(true);
      expect(wrapper.state('password1Error')).toBeNull();
      expect(wrapper.state('password2Error')).toBeNull();
      expect(wrapper.find(Redirect).exists()).toBe(true);
      expect(wrapper.find(Redirect).prop('to')).toBe('/accounts/login');
      expect(editUserUsername).not.toHaveBeenCalled();
      expect(editUserPassword).toHaveBeenCalledWith('password');
      done();
    });
  });

  test('errors on password mismatch', () => {
    const user = {
      user_id: 1,
      username: 'testuser',
      email: 'test@example.com',
      isSocial: false,
    };
    const wrapper = shallowWithIntl(
      <UserSetting
        user={user}
        editUserPassword={editUserPassword}
        editUserUsername={editUserUsername}
        fetchSubscription={fetchSubscription}
        refreshSession={refreshSession}
        upgradeSubscription={upgradeSubscription}
        isFetching
      />,
    ).dive().dive().dive();

    wrapper.find(TextField).at(1).simulate('change', {
      target: {
        name: 'password1',
        value: '1234',
      },
    });
    wrapper.find(TextField).at(2).simulate('change', {
      target: {
        name: 'password2',
        value: '5678',
      },
    });
    wrapper.update();
    wrapper.instance().saveUserPassword({ preventDefault: jest.fn() });

    expect(wrapper.state('saveSuccess')).toBe(false);
    expect(wrapper.state('password1Error')).toEqual(['Passwords must match']);
    expect(wrapper.state('password2Error')).toEqual(['Passwords must match']);
    expect(wrapper.find(Alert).exists()).toBe(true);
    expect(wrapper.find(Alert).last().prop('severity')).toBe('error');
    expect(editUserUsername).not.toHaveBeenCalled();
    expect(editUserPassword).not.toHaveBeenCalled();
  });

  test('displays error on username save error', (done) => {
    const user = {
      user_id: 1,
      username: 'testuser',
      email: 'test@example.com',
      isSocial: false,
    };
    const error = new Error();
    error.response = {
      status: 400,
      data: {
        username: ['Username taken'],
      },
    };
    editUserUsername = jest.fn().mockRejectedValue(error);
    const wrapper = shallowWithIntl(
      <UserSetting
        user={user}
        editUserPassword={editUserPassword}
        editUserUsername={editUserUsername}
        fetchSubscription={fetchSubscription}
        refreshSession={refreshSession}
        upgradeSubscription={upgradeSubscription}
        isFetching
      />,
    ).dive().dive().dive();

    wrapper.instance().saveUserUsername({ preventDefault: jest.fn() }).then(() => {
      expect(wrapper.state('saveSuccess')).toBe(false);
      expect(wrapper.state('usernameError')).toEqual(['Username taken']);
      expect(wrapper.find(Alert).exists()).toBe(true);
      expect(wrapper.find(Alert).last().prop('severity')).toBe('error');
      expect(wrapper.find(TextField).first().prop('error')).toBe(true);
      expect(wrapper.find(TextField).at(1).prop('error')).toBe(false);
      expect(wrapper.find(TextField).at(2).prop('error')).toBe(false);
      done();
    });
  });

  test('displays error on password save error', (done) => {
    const user = {
      user_id: 1,
      username: 'testuser',
      email: 'test@example.com',
      isSocial: false,
    };
    const error = new Error();
    error.response = {
      status: 400,
      data: {
        new_password1: ['Too short'],
        new_password2: ['Too short'],
      },
    };
    editUserPassword = jest.fn().mockRejectedValue(error);
    const wrapper = shallowWithIntl(
      <UserSetting
        user={user}
        editUserPassword={editUserPassword}
        editUserUsername={editUserUsername}
        fetchSubscription={fetchSubscription}
        refreshSession={refreshSession}
        upgradeSubscription={upgradeSubscription}
        isFetching
      />,
    ).dive().dive().dive();

    wrapper.instance().saveUserPassword({ preventDefault: jest.fn() }).then(() => {
      expect(wrapper.state('saveSuccess')).toBe(false);
      expect(wrapper.state('password1Error')).toEqual(['Too short']);
      expect(wrapper.state('password2Error')).toEqual(['Too short']);
      expect(wrapper.find(Alert).exists()).toBe(true);
      expect(wrapper.find(Alert).last().prop('severity')).toBe('error');
      expect(wrapper.find(TextField).first().prop('error')).toBe(false);
      expect(wrapper.find(TextField).at(1).prop('error')).toBe(true);
      expect(wrapper.find(TextField).at(2).prop('error')).toBe(true);
      done();
    });
  });

  test('displays plans', () => {
    const user = {
      user_id: 1,
      username: 'testuser',
      email: 'test@example.com',
      isSocial: false,
    };
    const subscription = {
      plan: '2',
      price: 0,
      interval: '',
      start: 0,
    };
    const wrapper = shallowWithIntl(
      <UserSetting
        user={user}
        subscription={subscription}
        editUserPassword={editUserPassword}
        editUserUsername={editUserUsername}
        fetchSubscription={fetchSubscription}
        refreshSession={refreshSession}
        upgradeSubscription={upgradeSubscription}
        isFetching={false}
      />,
    ).dive().dive().dive();
    expect(wrapper.find(PlanItem).length).toBe(3);
    expect(wrapper.find(PlanItem).at(0).prop('active')).toBe(false);
    expect(wrapper.find(PlanItem).at(1).prop('active')).toBe(false);
    expect(wrapper.find(PlanItem).at(2).prop('active')).toBe(true);
  });

  test('handles upgrade error', () => {
    const user = {
      user_id: 1,
      username: 'testuser',
      email: 'test@example.com',
      isSocial: false,
    };
    const subscription = {
      plan: '1',
      price: 0,
      interval: '',
      start: 0,
    };
    const wrapper = shallowWithIntl(
      <UserSetting
        user={user}
        subscription={subscription}
        editUserPassword={editUserPassword}
        editUserUsername={editUserUsername}
        fetchSubscription={fetchSubscription}
        refreshSession={refreshSession}
        upgradeSubscription={upgradeSubscription}
        isFetching={false}
        upgradeError={{
          response: {
            data: {
              accessCode: ['This code is not valid.'],
            },
          },
        }}
      />,
    ).dive().dive().dive();
    expect(wrapper.find(PlanItem).length).toBe(3);
    expect(wrapper.find(PlanItem).at(0).prop('active')).toBe(true);
    expect(wrapper.find(PlanItem).at(1).prop('active')).toBe(false);
    expect(wrapper.find(PlanItem).at(2).prop('active')).toBe(false);
    expect(wrapper.find(PlanClass).prop('error')).toBe('This code is not valid.');
  });
});
