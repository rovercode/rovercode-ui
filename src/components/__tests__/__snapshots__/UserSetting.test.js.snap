// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The UserSetting component renders on the page with no errors 1`] = `
<WithStyles(ForwardRef(Container))
  className="UserSetting-mainContainer-1"
>
  <WithStyles(ForwardRef(Grid))
    className="UserSetting-settingsSection-2"
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      className="UserSetting-titleArea-4"
      container={true}
      direction="row"
      item={true}
      justify="space-between"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Typography))
          className="UserSetting-title-3"
          variant="h4"
        >
          <FormattedMessage
            defaultMessage="Account"
            id="app.user_setting.header"
          />
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={2}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Alert))
          severity="info"
        >
          <FormattedMessage
            defaultMessage="Changing any of these settings requires signing back in."
            id="app.user_setting.info"
          />
        </WithStyles(ForwardRef(Alert))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={4}
      xs={12}
    >
      <Styled(MuiBox)
        border={1}
        borderColor="grey.500"
        borderRadius="borderRadius"
        boxShadow={3}
        p={3}
      >
        <WithStyles(ForwardRef(Grid))
          alignItems="center"
          container={true}
          direction="row"
          item={true}
          justify="center"
          spacing={1}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <PersonIcon
              fontSize="medium"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <WithStyles(ForwardRef(Typography))
              variant="h6"
            >
              <FormattedMessage
                defaultMessage="Change Username"
                id="app.user_setting.username_header"
              />
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
        <form
          key="1"
          onSubmit={[Function]}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <WithStyles(ForwardRef(TextField))
              defaultValue="testuser"
              error={false}
              label="Username:"
              margin="dense"
              name="username"
              onChange={[Function]}
              required={true}
              variant="outlined"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <WithStyles(ForwardRef(Button))
              color="primary"
              type="submit"
              variant="contained"
            >
              <FormattedMessage
                defaultMessage="Save"
                id="app.user_setting.save_username"
              />
            </WithStyles(ForwardRef(Button))>
          </WithStyles(ForwardRef(Grid))>
        </form>
      </Styled(MuiBox)>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={6}
      xs={12}
    >
      <Styled(MuiBox)
        border={1}
        borderColor="grey.500"
        borderRadius="borderRadius"
        boxShadow={3}
        p={3}
      >
        <WithStyles(ForwardRef(Grid))
          alignItems="center"
          container={true}
          direction="row"
          item={true}
          justify="center"
          spacing={1}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <LockIcon
              fontSize="medium"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <WithStyles(ForwardRef(Typography))
              variant="h6"
            >
              <FormattedMessage
                defaultMessage="Change Password"
                id="app.user_setting.password_header"
              />
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
        <form
          key="1"
          onSubmit={[Function]}
        >
          <WithStyles(ForwardRef(Grid))
            container={true}
            direction="row"
            item={true}
            spacing={1}
          >
            <WithStyles(ForwardRef(Grid))
              item={true}
            >
              <WithStyles(ForwardRef(TextField))
                error={false}
                label="New Password:"
                margin="dense"
                name="password1"
                onChange={[Function]}
                required={true}
                type="password"
                variant="outlined"
              />
            </WithStyles(ForwardRef(Grid))>
            <WithStyles(ForwardRef(Grid))
              item={true}
            >
              <WithStyles(ForwardRef(TextField))
                error={false}
                label="Verify:"
                margin="dense"
                name="password2"
                onChange={[Function]}
                required={true}
                type="password"
                variant="outlined"
              />
            </WithStyles(ForwardRef(Grid))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <WithStyles(ForwardRef(Button))
              color="primary"
              type="submit"
              variant="contained"
            >
              <FormattedMessage
                defaultMessage="Save"
                id="app.user_setting.save_password"
              />
            </WithStyles(ForwardRef(Button))>
          </WithStyles(ForwardRef(Grid))>
        </form>
      </Styled(MuiBox)>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    className="UserSetting-settingsSection-2"
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      className="UserSetting-titleArea-4"
      container={true}
      direction="row"
      item={true}
      justify="space-between"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Typography))
          className="UserSetting-title-3"
          variant="h5"
        >
          <FormattedMessage
            defaultMessage="Plans"
            id="app.user_setting.plans"
          />
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="flex-start"
      container={true}
      direction="row"
      item={true}
      spacing={1}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Skeleton))
          animation="wave"
          height={200}
          width={300}
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Skeleton))
          animation="wave"
          height={200}
          width={300}
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Skeleton))
          animation="wave"
          height={200}
          width={300}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Container))>
`;
